src/pyats/contrib/creators/topology.py

```diff
@@ -0,0 +1,656 @@
+        '''
+        new_entry = {'dest_host': dest_host, 
+                    'dest_port': dest_port, 
+                    'ip_address': ip_address}
+        if interface not in connection_dict:
+            connection_dict[interface] = [new_entry]
+        else:
+            unique = True
+            for entry in connection_dict[interface]:
+
+                # check that the connection being added is unique
+                if (entry['dest_host'] == dest_host 
+                        and entry['dest_port'] == dest_port):
+                    unique = False
+                    break
+            if unique:
+                log.info('Connection device {} interface {} to' 
+                         ' device {} interface {} found'.format(dev, 
+                                                                interface, 
+                                                                dest_host, 
+                                                                dest_port))
+                connection_dict[interface].append(new_entry)
+                
+    def get_credentials_and_proxies(self, testbed):
+        '''
+        Takes a copy of the current credentials in the testbed 
+        for use in connecting to other devices
+        '''
+        credential_dict = {}
+        proxy_list = set()
+        for device in testbed['devices'].values():
+
+            # get all connections used in the testbed
+            if 'credentials' not in device:
+                continue
+            for cred in device['credentials']:
+                if cred not in credential_dict :
+                    credential_dict[cred] = dict(device['credentials'][cred])
+                elif device['credentials'][cred] not in credential_dict.values():
+                    credential_dict[cred + str(len(credential_dict))] = dict(device['credentials'][cred])
+
+            #get list of proxies used in connections
+            for connect in device['connections'].values():
+                if 'proxy' in connect:
+                    proxy_list.add(connect['proxy'])
+
+        return credential_dict, proxy_list
+
+    def get_os(self, system_string, platform_name):
+        ''' 
+        get the os from the system_description output from the show
+        cdp and show lldp neighbor parsers
+        '''
+        if 'IOS' in system_string or 'IOS' in platform_name:
+            if 'XE' in system_string or 'XE' in platform_name:
+                return 'iosxe'
+            elif 'XR' in system_string or 'XR' in platform_name:
+                return 'iosxr'
+            else:
+                return 'ios'
+        if 'NX-OS' in system_string or 'NX-OS' in platform_name:
+            return 'nxos'
+
+    def get_interfaces_ip_address(self, device):
+        '''
+        get the ip address for all of the generated interfaces
+        on the give device
+        '''
+        if (not device.is_connected() or device.os not in os_list 

```

Comment:
no need of the ( ) around that if statement

Reply:
removed brackets